community/kernel/src/main/java/org/neo4j/graphdb/index/IndexImplementation.java

```diff
@@ -35,6 +35,9 @@
  */
 public interface IndexImplementation extends Lifecycle
 {
+
+    File getStoreDirectory( File storeDir );
```

Comment:
Javadocs please. You're overriding the term "store directory", at least explain why.